code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -231,7 +233,63 @@
 	if(R.hud_used)
 		R.hud_used.update_robot_modules_display()
 	SSblackbox.record_feedback("tally", "cyborg_modules", 1, R.module)
-
+	
+/obj/item/robot_module/proc/analyze_reagents(var/list/reagents_inside)
+	var/mob/living/silicon/robot/R = loc
+	to_chat(loc, "<span class='notice'>You start analyzing the solution.</span>")
+	var/do_what = 0
+	var/analyzed = FALSE
+	var/i = 0
+	for(var/datum/reagent/each_reagent in reagents_inside) //If there are multiple reagents
+		do_what = 0 //Resets on each reagent
+		analyzed = FALSE
+		i += 10 //So it kinda prevents spam
+		if(each_reagent in learned_reagents) //If you can already synthetize it, no need to analyze again
+			analyzed = TRUE 
+		else
+			R.cell.use(15) //Consumes energy to analyze. Not sure what happens if the borg do not have enough energy. Will consume energy even if it fails to analyze
+			if(!each_reagent.can_synth)
+				do_what = 1
+			else if(each_reagent in whitelisted_reagents || R.emagged || !whitelisted_reagents.len) //Only allowed reagents. Can be set by borg type. Except if emmaged. If empty then allow everything
+				do_what = 2
+			else
+				do_what = 3

```

Comment:
this

Reply:
Should I change them to words? I mean.. it can work too

Reply:
sure if that's not going to be problematic.